src/main/java/team9/ddang/member/repository/FriendRepository.java

```diff
@@ -0,0 +1,31 @@
+package team9.ddang.member.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import team9.ddang.member.entity.Friend;
+import team9.ddang.member.entity.Member;
+
+import java.util.List;
+
+public interface FriendRepository  extends JpaRepository<Friend, Long> {
+
+    @Query("""
+    SELECT f.receiver
+    FROM Friend f
+    WHERE f.sender = :sender
+    and f.isDeleted = 'FALSE'
+    and f.receiver.isDeleted = 'FALSE'
+    """)
+    List<Member> findAllFriendsBySender(Member sender);
+
+    @Modifying
+    @Query("""
+    DELETE FROM Friend f
+    WHERE f.sender = :sender 
+    AND f.receiver = :receiver
+    """)
+    void deleteBySenderAndReceiver(Member sender, Member receiver);
```

Comment:
Service 부분에서 deleteFriend에서 2번 호출하는 거 보고 왔는데
쿼리를 수정해서 한번만 호출하는 게 더 좋아보입니다.
```suggestion
    @Modifying
    @Query("""
    DELETE FROM Friend f
    WHERE (f.sender = :member AND f.receiver = :otherMember)
       OR (f.sender = :otherMember AND f.receiver = :member)
""")
    void deleteBySenderAndReceiver(Member member, Member otherMember);
```

Reply:
감사합니다! 지금 그냥 깊게 생각을 못하고 짜는 거 같은데 이런 부분 잡아주시는 거 감사하네요..